### PR TITLE
message when sharing URL for private

### LINKED ISSUE #2793: Downloading file via URL fails if it isn't public
## Steps to Reproduce
1. create hs resource, keep sharing status private
2. right click on file --> `Get file URL`
3. in local terminal download with `wget`: wget <file url>

This gives a 403 forbidden error, which is expected since HS cannot verify who is downloading the resource.  

## Suggested Fix

1. Deactivate `Get file URL` button for non-public resources
2. Tell user that resource must be public to access via file url

### PR DESCRIPTION
**Pull request by devincowan**
### Pull Request Checklist: 
- [x] Positive Test Case Written by Dev

- [ ] Automated Testing

- [ ] Sufficient User and Developer Documentation

- [ ] Passing Jenkins Build

- [ ] Peer Code review and approval

### Positive Test Case
1. Manage who has access on a resource that has adequate metadata to be made public
2. First set the access to private
3. Within the file-browser-container, upload a file if the resource doesn't already have one
4. Right click the file -> "get file url"
5. You will see a message in the modal explaining that the resource is private
6. Now change the "manage access" to discoverable and "Enable private link sharing"
7. Right click the same file -> "get file url"
8. You see a message explaining that the resource is discoverable with link sharing

Resolves #2793

***
#### Before:
41728169-58f054a2-7533-11e8-8800-b6fbddffd47a

#### After:
Screen Shot 2022-03-03 at 2 46 43 PM

### CODE REVIEW

**Comment by hydrocheck** [comment]
| METRIC | VALUE |
|:-----:|:-----:|

[https://sonarqube.cuahsi-workstation.com:9000/dashboard?id=hydroshare-4522](https://sonarqube.cuahsi-workstation.com:9000/dashboard?id=hydroshare-4522)

File: theme/templates/resource-landing-page/modals.html [hunk 1]
```diff
@@ -1775,6 +1775,33 @@ <h4 class="modal-title">Get file URL</h4>
             </div>
 
             <div class="modal-body">
+                <div id="warnTxtFileURL" class="row" style="display:none">
```
Comment (Maurier) [inline theme/templates/resource-landing-page/modals.html:1778]: I would move this inside the div below and give it the class `'col-sm-12'` so we get consisting padding. Typically, for layouts like this, you only need one `row` with its `col` children, unless it makes sense structurally to separate them.

**Review by Maurier — CHANGES_REQUESTED** [review]
Looks good. Just minor layout suggestions.

File: theme/templates/resource-landing-page/modals.html (hunk 1 above)
Comment (devincowan) [reply theme/templates/resource-landing-page/modals.html]: makes sense, thanks. Just pushed a commit to make this change

**Review by Maurier — APPROVED** [review]

**Comment by Maurier** [comment]
@devincowan Also, we could use the styling here in order to be consistent with notifications like this.

image

But I will leave it up to you and other members of the team.

**Comment by devincowan** [comment]
thanks @Maurier good suggestion, I updated the style in [4bb2888](https://github.com/hydroshare/hydroshare/pull/4522/commits/4bb2888aaa2777043e5b78d1ff9408dece2b5c60) so this should be all set

**Review by Maurier — CHANGES_REQUESTED** [review]
Looks good, but the padding is off:

image

**Comment by devincowan** [comment]
good catch thank you fixed in [347c991](https://github.com/hydroshare/hydroshare/pull/4522/commits/347c9915f66844ae691037b5c65ea8a57cf9e358)
image

**Review by Maurier — APPROVED** [review]
Looks good

**Pull request merged**